blockindex/contractstaking/delta_cache.go

```diff
@@ -68,84 +68,65 @@ func (s *contractStakingDelta) MatchBucketType(amount *big.Int, duration uint64)
 }
 
 func (s *contractStakingDelta) GetBucketInfo(id uint64) (*bucketInfo, deltaState) {
-	if state, ok := s.bucketInfoDeltaState[id]; ok {
-		switch state {
-		case deltaStateAdded, deltaStateModified:
-			return s.cache.MustGetBucketInfo(id), state
-		default:
-			return nil, state
-		}
+	state := s.bucketInfoDeltaState[id]
+	switch state {
+	case deltaStateAdded, deltaStateModified:
+		return s.cache.MustGetBucketInfo(id), state
+	default: // deltaStateRemoved, deltaStateUnchanged
+		return nil, state
 	}
-	return nil, deltaStateReverted
 }
 
 func (s *contractStakingDelta) GetBucketType(id uint64) (*BucketType, deltaState) {
-	if state, ok := s.bucketTypeDeltaState[id]; ok {
-		switch state {
-		case deltaStateAdded, deltaStateModified:
-			return s.cache.MustGetBucketType(id), state
-		default:
-			return nil, state
-		}
+	state := s.bucketTypeDeltaState[id]
+	switch state {
+	case deltaStateAdded, deltaStateModified:
+		return s.cache.MustGetBucketType(id), state
+	default: // deltaStateUnchanged
+		return nil, state
 	}
-	return nil, deltaStateReverted
 }
 
 func (s *contractStakingDelta) AddBucketInfo(id uint64, bi *bucketInfo) error {
 	return s.addBucketInfo(id, bi)
 }
 
 func (s *contractStakingDelta) AddBucketType(id uint64, bt *BucketType) error {
-	if _, ok := s.bucketTypeDeltaState[id]; !ok {
-		s.bucketTypeDeltaState[id] = deltaStateAdded
-	} else {
-		var err error
-		s.bucketTypeDeltaState[id], err = s.bucketTypeDeltaState[id].Transfer(deltaActionAdd)
-		if err != nil {
-			return err
-		}
+	var err error
+	s.bucketTypeDeltaState[id], err = s.bucketTypeDeltaState[id].Transfer(deltaActionAdd)
+	if err != nil {
+		return err
```

Comment:
interesting programming, and this is banking on the fact that 
1. an uninitialized`bucketTypeDeltaState[id]` returns the zero-value of int which is 0
2. and `deltaStateUnchanged` also equals 0 in value

may sound stupid, but let's add a test to guarantee the above 2 points

Reply:
panic if bucketTypeDeltaState[id] not exist?

Reply:
we should do check `bucketTypeDeltaState[id]` is nil

Reply:
In Go, accessing a map with a non-existent key will return the zero value for the type of the value in the map. In this case, it will return `deltaStateUnchanged`

Reply:
already added a zero-value unit test